### ISSUE TITLE
Lint against format_implicit_args

### ISSUE THREAD

**Issue by dylni** [rust-lang/rust-clippy#5404]
Would you consider a lint to warn for uses of the "format_implicit_args" feature now that rust-lang/rfcs#2795 has been merged?

As far as I'm aware it's the only way the standard library provides for a string to be interpreted as a variable. It can be done using procedural macros, but that would require adding a crate as a dependency. So, I would prefer a lint against this, because I believe it to be an anti-pattern.

**Comment by flip1995** [comment]
You mean a lint, that fires on every use of implicit args?

**Comment by dylni** [comment]
Yes. If clippy stops after the first usage, that would be acceptable, but I haven't found that to be how it typically behaves.

**Comment by flip1995** [comment]
This would be linting on a whole feature, that got accepted through an RFC, which Clippy normally doesn't do.

But Clippy has the `restriction` lint group, where such a lint would be possible, but then has to be enabled explicitly. This lint would first require this feature to be implemented in rustc though.

**Comment by dylni** [comment]
The restriction group definitely makes the most sense.

I figured it would need to be implemented in rustc first, but I wanted to create the issue now to see if this would even be considered with the RFC being accepted. There was [some opposition](https://github.com/rust-lang/rfcs/pull/2795#issuecomment-551260362) on the RFC though, which also showed in the voting, so I think this could be generally useful.

**Comment by MightyPork** [comment]
I would actually be interested in the opposite lint of the proposal here, a lint warning about the old syntax in format macros. It could even have automatic fix available.

**Comment by dylni** [comment]
This could be similar to [implicit\_return](https://rust-lang.github.io/rust-clippy/master/index.html#implicit_return) and [needless\_return](https://rust-lang.github.io/rust-clippy/master/index.html#needless_return), which both exist for both styles.

Also, this issue can now be unblocked.

@rustbot labels: -S-blocked
